presto-main/src/main/java/io/prestosql/metadata/OperatorNotFoundException.java

```diff
@@ -52,17 +52,34 @@ public OperatorNotFoundException(OperatorType operatorType, List<? extends Type>
 
     private static String formatErrorMessage(OperatorType operatorType, List<? extends Type> argumentTypes, Optional<TypeSignature> returnType)
     {
-        String operatorString;
         switch (operatorType) {
+            case ADD:
+            case SUBTRACT:
+            case MULTIPLY:
+            case DIVIDE:
+            case MODULUS:
+            case EQUAL:
+            case NOT_EQUAL:
+            case LESS_THAN:
+            case LESS_THAN_OR_EQUAL:
+            case GREATER_THAN:
+            case GREATER_THAN_OR_EQUAL:
+                return format("Cannot apply operator: %s %s %s", argumentTypes.get(0), operatorType.getOperator(), argumentTypes.get(1));
+            case NEGATION:
+                return format("Cannot negate %s", argumentTypes.get(0));
             case BETWEEN:
                 return format("Cannot check if %s is BETWEEN %s and %s", argumentTypes.get(0), argumentTypes.get(1), argumentTypes.get(2));
+            case IS_DISTINCT_FROM:
+                return format("Cannot check if %s IS DISTINCT FROM %s", argumentTypes.get(0), argumentTypes.get(1));
```

Comment:
lowercase IS DISTINCT FROM ?
(you didn't upper cast in the CAST case; and the sentence is still a sentence when you treat "is distinct from" as just words)

Reply:
I was following BETWEEN, but I'll change that as well.

Reply:
Actually, this matches what is used in `ExpressionAnalyzer`, so I'll make it `DISTINCT FROM`.